gnome-initial-setup/pages/timezone/gis-timezone-page.c

```diff
@@ -455,11 +496,18 @@ gis_timezone_page_dispose (GObject *object)
   GisTimezonePage *page = GIS_TIMEZONE_PAGE (object);
   GisTimezonePagePrivate *priv = gis_timezone_page_get_instance_private (page);
 
-  stop_geolocation (page);
+  stop_geolocation_if_needed (page);
 
   g_clear_object (&priv->dtm);
   g_clear_object (&priv->clock);
 
+  if (priv->network_monitor != NULL && priv->network_monitor_handler_id > 0) {
+    g_signal_handler_disconnect (priv->network_monitor,
+                                 priv->network_monitor_handler_id);
+    priv->network_monitor_handler_id = 0;
+  }
+  g_clear_object (&priv->network_monitor);
```

Comment:
`g_network_monitor_get_default()` does not return a new reference. In fact there should be no need for you to even store it in the private pointer here, since it's already a singleton.